### PR TITLE
get absolute IN_FILE based on SOURCE_DIR of DEPENDS_TRGT

### PR DESCRIPTION
**Pull request by tandf**

### CODE REVIEW

File: cmake/LLVMIRUtil.cmake [hunk 1]
```diff
@@ -175,16 +175,17 @@ function(llvmir_attach_bc_target)
       continue()
     endif()
 
+    get_target_property(SOURCE_DIR ${DEPENDS_TRGT} SOURCE_DIR)
+    get_filename_component(INFILE ${IN_FILE} ABSOLUTE BASE_DIR ${SOURCE_DIR})
```
Comment (compor) [inline cmake/LLVMIRUtil.cmake:179]: Can we unify the use of `INFILE` with `IN_FILE`?
I find the current naming confusing.

File: cmake/LLVMIRUtil.cmake [hunk 2]
```diff
@@ -175,16 +175,17 @@ function(llvmir_attach_bc_target)
...
     get_filename_component(OUTFILE ${IN_FILE} NAME)
```
Comment (compor) [inline cmake/LLVMIRUtil.cmake:180]: What would happen if you used the `INFILE`  here? Would we get the same thing?

File: cmake/LLVMIRUtil.cmake [hunk 3]
```diff
@@ -175,16 +175,17 @@ function(llvmir_attach_bc_target)
...
-    get_filename_component(INFILE ${IN_FILE} ABSOLUTE)
     set(OUT_LLVMIR_FILE "${OUTFILE}.${LLVMIR_BINARY_FMT_SUFFIX}")
     set(FULL_OUT_LLVMIR_FILE "${WORK_DIR}/${OUT_LLVMIR_FILE}")
 
     # compile definitions per source file
-    llvmir_extract_compile_defs_properties(IN_FILE_DEFS ${IN_FILE})
+    llvmir_extract_compile_defs_properties(IN_FILE_DEFS ${INFILE})
```
Comment (compor) [inline cmake/LLVMIRUtil.cmake:185]: Ideally, I'd like to use only one version of the "infile" as derived from any path processing of `IN_FILE` which changes at the `foreach`.
Can this be done?
Maybe name it `ABS_IN_FILE` or something similar?

**Review by compor — CHANGES_REQUESTED** [review]

File: cmake/LLVMIRUtil.cmake [hunk 4]
```diff
@@ -175,16 +175,17 @@ function(llvmir_attach_bc_target)
       continue()
     endif()
 
+    get_target_property(SOURCE_DIR ${DEPENDS_TRGT} SOURCE_DIR)
+    get_filename_component(INFILE ${IN_FILE} ABSOLUTE BASE_DIR ${SOURCE_DIR})
     get_filename_component(OUTFILE ${IN_FILE} NAME)
-    get_filename_component(INFILE ${IN_FILE} ABSOLUTE)
```
Comment (tandf) [inline cmake/LLVMIRUtil.cmake:179]: Hi compor, I was trying to follow the original naming pattern here, so I chose to maintain both `INFILE` and `IN_FILE`. Do you think it's ok if I change all `INFILE` to `ABS_IN_FILE` as you proposed?

File: cmake/LLVMIRUtil.cmake (hunk 2 above)
Comment (tandf) [reply cmake/LLVMIRUtil.cmake:180]: Yes, I think so. According to [CMake's document](https://cmake.org/cmake/help/latest/command/get_filename_component.html), `get_filename_component(<var> <FileName> NAME)` returns "File name without directory". Therefore, we should get the same result either passing a relative path or a absolute path.

File: cmake/LLVMIRUtil.cmake (hunk 4 above)
Comment (compor) [reply cmake/LLVMIRUtil.cmake:179]: Yeah, but I'd say use `ABS_IN_FILE` once it is an absolute path to a file.

File: cmake/LLVMIRUtil.cmake (hunk 2 above)
Comment (compor) [reply cmake/LLVMIRUtil.cmake:180]: Maybe try it to make sure and then you can assign it to the variable with a name that indicates an absolute file path as we discussed below.

Comment (tandf) [reply cmake/LLVMIRUtil.cmake]: I just tried, and the results are the same when using `IN_FILE` or the absolute version `ABS_IN_FILE`

File: cmake/LLVMIRUtil.cmake [hunk 5]
```diff
@@ -166,25 +166,27 @@ function(llvmir_attach_bc_target)
 
   # main operations
   foreach(IN_FILE ${IN_FILES})
+    get_target_property(SOURCE_DIR ${DEPENDS_TRGT} SOURCE_DIR)
+    get_filename_component(ABS_IN_FILE ${IN_FILE} ABSOLUTE BASE_DIR ${SOURCE_DIR})
```
Comment (tandf) [inline cmake/LLVMIRUtil.cmake:170]: Get the absolute path at the start of each loop, and replace all usage of `IN_FILE` and `INFILE` with `ABS_IN_FILE`.

According to the document and some simple tests, the results are the same when using `IN_FILE` and `ABS_IN_FILE`, either for getting ext `get_filename_component(FILE_EXT ${ABS_IN_FILE} LAST_EXT)` or for getting file name `get_filename_component(OUTFILE ${ABS_IN_FILE} NAME)`.

Comment (compor) [reply cmake/LLVMIRUtil.cmake:170]: LGTM, please combine the commits and I'll merge it in.

**Review by compor — APPROVED** [review]
LGTM

**Pull request merged**